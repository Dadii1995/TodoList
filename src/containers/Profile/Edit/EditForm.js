import React, { useState } from 'react'
import { Field, Form, Formik } from 'formik'
import { Alert, Button } from 'reactstrap'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import * as Yup from 'yup'

import TextInput from '../../../components/Form/TextInput'
import DatePicker from '../../../components/Form/DatePicker'

const formSchema = Yup.object().shape({
  firstName: Yup.string().required('Required'),
  url: Yup.string().url('Photo url must be url'),
  birthday: Yup.date(),
})

const EditForm = props => {
  const [profile] = useState(props.profile)
  return (
    <Formik
      initialValues={{ ...profile }}
      onSubmit={props.saveProfile}
      validationSchema={formSchema}
    >
      {({ isSubmitting, errors }) => (
        <Form>
          <Field
            component={TextInput}
            dataCy="user-name-input"
            label="First name"
            name="firstName"
          />
          {errors.firstName ? <Alert color="danger">{errors.firstName}</Alert> : null}
          <Field
            component={TextInput}
            dataCy="user-photo-input"
            label="Photo url"
            name="url"
            placeholder="http://examplewebsite.com/img/1.png"
          />
          {errors.url ? <Alert color="danger">{errors.url}</Alert> : null}

          <Field
            component={DatePicker}
            dataCy="user-birthday-input"
            label="Birthday"
            name="birthday"
          />
          {errors.birthday ? <Alert color="danger">{errors.birthday}</Alert> : null}

          <Button
            block
            color="success"
            data-cy="submit-button"
            disabled={isSubmitting}
            type="submit"
          >
            Save
          </Button>
        </Form>
      )}
    </Formik>
  )
}
EditForm.propTypes = {
  saveProfile: PropTypes.func,
  profile: PropTypes.object,
}
const mapStateToProps = ({ profile }) => ({
  profile,
})
export default connect(
  mapStateToProps,
  undefined,
)(EditForm)
