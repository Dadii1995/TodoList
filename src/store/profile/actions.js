import { EDIT_PROFILE, CHANGE_PASSWORD } from './types'

export const editProfile = ({ firstName, url, birthday }) => {
  return {
    type: EDIT_PROFILE,
    payload: {
      firstName,
      url,
      birthday,
    },
  }
}

export const changePassword = password => ({
  type: CHANGE_PASSWORD,
  payload: password,
})
