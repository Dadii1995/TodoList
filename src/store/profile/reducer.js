import { EDIT_PROFILE, CHANGE_PASSWORD } from './types'

export const initialState = JSON.parse(
  localStorage.getItem('profile') ||
    '{"firstName":"NewUser","url":"","password":"admin123","birthday":"1995-06-23"}',
) || {
  firstName: 'NewUser',
  url: '',
  password: 'admin123',
  birthday: '1995-06-23',
}
const profile = (state = initialState, { type, payload }) => {
  switch (type) {
    case EDIT_PROFILE: {
      const updatedState = {
        ...state,
        ...payload,
      }
      localStorage.setItem('profile', JSON.stringify(updatedState))
      return updatedState
    }
    case CHANGE_PASSWORD: {
      const updatedState = {
        ...state,
        password: payload,
      }
      localStorage.setItem('profile', JSON.stringify(updatedState))
      return updatedState
    }
    default:
      return state
  }
}
export default profile
