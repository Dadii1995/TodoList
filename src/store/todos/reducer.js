import { ADD_TODO, GET_TODO_BY_ID, TOGGLE_TODO } from './types'

export const initialState = JSON.parse(localStorage.getItem('todos') || '[]') || []
const todos = (state = initialState, { type, payload }) => {
  switch (type) {
    case ADD_TODO: {
      const updatedState = [...state, { ...payload, isDone: false }]
      localStorage.setItem('todos', JSON.stringify(updatedState))
      return updatedState
    }
    case TOGGLE_TODO: {
      const updatedState = state.map(
        todo => (todo.id === payload ? { ...todo, isDone: !todo.isDone } : todo),
      )
      localStorage.setItem('todos', JSON.stringify(updatedState))
      return updatedState
    }
    case GET_TODO_BY_ID:
      return state.filter(todo => todo.id === payload)

    default:
      return state
  }
}

export default todos
