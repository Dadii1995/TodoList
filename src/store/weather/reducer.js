import { GET_WEATHER, SET_WEATHER_LOADING, SET_WEATHER_ERROR, SET_WEATHER_CITY } from './types'

export const initialState = {
  current: { condition: {} },
  location: {},
  isLoading: false,
  error: '',
  city: 'Bielsko-Biala',
}

export const weather = (state = initialState, action) => {
  switch (action.type) {
    case GET_WEATHER: {
      return { ...state, ...action.weather }
    }
    case SET_WEATHER_CITY: {
      return { ...state, city: action.payload }
    }
    case SET_WEATHER_LOADING: {
      return { ...state, isLoading: action.payload }
    }
    case SET_WEATHER_ERROR: {
      return { ...state, error: action.payload }
    }
    default:
      return state
  }
}
export default weather
