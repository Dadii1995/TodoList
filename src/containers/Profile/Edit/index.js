import React from 'react'
import EditForm from './EditForm'
import { editProfile } from '../../../store/profile/actions'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

const Edit = props => {
  const saveProfile = (values, { setSubmitting }) => {
    setTimeout(() => {
      props.editProfile(values)
      props.history.push('/profile')
      setSubmitting(false)
    }, 500)
  }
  return <EditForm saveProfile={saveProfile} />
}
const mapDispatchToProps = {
  editProfile,
}

Edit.propTypes = {
  editProfile: PropTypes.func,
  history: PropTypes.object,
}

export default connect(
  undefined,
  mapDispatchToProps,
)(Edit)
