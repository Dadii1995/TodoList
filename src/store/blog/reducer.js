import {
  GET_POSTS,
  DELETE_SELECTED_POSTS,
  SELECT_POST,
  SET_POSTS_ERROR,
  SET_POSTS_LOADING,
  ORDER_BY,
  FILTER,
} from './types'
import sortingAndFilteringPosts from '../../utils/sortingAndFilteringPosts'
import { OrderTypes } from './actions'

export const initialState = {
  posts: [
    {
      userId: 1,
      id: 1,
      title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
      body:
        'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
    },
    {
      userId: 1,
      id: 2,
      title: 'qui est esse',
      body:
        'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
    },
    {
      userId: 1,
      id: 3,
      title: 'ea molestias quasi exercitationem repellat qui ipsa sit aut',
      body:
        'et iusto sed quo iure\nvoluptatem occaecati omnis eligendi aut ad\nvoluptatem doloribus vel accusantium quis pariatur\nmolestiae porro eius odio et labore et velit aut',
    },
    {
      userId: 1,
      id: 4,
      title: 'eum et est occaecati',
      body:
        'ullam et saepe reiciendis voluptatem adipisci\nsit amet autem assumenda provident rerum culpa\nquis hic commodi nesciunt rem tenetur doloremque ipsam iure\nquis sunt voluptatem rerum illo velit',
    },
    {
      userId: 1,
      id: 5,
      title: 'nesciunt quas odio',
      body:
        'repudiandae veniam quaerat sunt sed\nalias aut fugiat sit autem sed est\nvoluptatem omnis possimus esse voluptatibus quis\nest aut tenetur dolor neque',
    },
    {
      userId: 1,
      id: 6,
      title: 'dolorem eum magni eos aperiam quia',
      body:
        'ut aspernatur corporis harum nihil quis provident sequi\nmollitia nobis aliquid molestiae\nperspiciatis et ea nemo ab reprehenderit accusantium quas\nvoluptate dolores velit et doloremque molestiae',
    },
  ],
  filteredPosts: [
    {
      userId: 1,
      id: 1,
      title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
      body:
        'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
    },
    {
      userId: 1,
      id: 2,
      title: 'qui est esse',
      body:
        'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
    },
    {
      userId: 1,
      id: 3,
      title: 'ea molestias quasi exercitationem repellat qui ipsa sit aut',
      body:
        'et iusto sed quo iure\nvoluptatem occaecati omnis eligendi aut ad\nvoluptatem doloribus vel accusantium quis pariatur\nmolestiae porro eius odio et labore et velit aut',
    },
    {
      userId: 1,
      id: 4,
      title: 'eum et est occaecati',
      body:
        'ullam et saepe reiciendis voluptatem adipisci\nsit amet autem assumenda provident rerum culpa\nquis hic commodi nesciunt rem tenetur doloremque ipsam iure\nquis sunt voluptatem rerum illo velit',
    },
    {
      userId: 1,
      id: 5,
      title: 'nesciunt quas odio',
      body:
        'repudiandae veniam quaerat sunt sed\nalias aut fugiat sit autem sed est\nvoluptatem omnis possimus esse voluptatibus quis\nest aut tenetur dolor neque',
    },
    {
      userId: 1,
      id: 6,
      title: 'dolorem eum magni eos aperiam quia',
      body:
        'ut aspernatur corporis harum nihil quis provident sequi\nmollitia nobis aliquid molestiae\nperspiciatis et ea nemo ab reprehenderit accusantium quas\nvoluptate dolores velit et doloremque molestiae',
    },
  ],
  isLoading: false,
  error: '',
  selectedPosts: [2, 3],
  orderBy: OrderTypes.DATE_ASC,
  filter: '',
}

export const blog = (state = initialState, { type, payload }) => {
  switch (type) {
    case GET_POSTS:
      return {
        ...state,
        orderBy: OrderTypes.DATE_ASC,
        filter: '',
        posts: payload,
        filteredPosts: payload,
      }
    case DELETE_SELECTED_POSTS:
      return {
        ...state,
        posts: state.posts.filter(post => !state.selectedPosts.includes(post.id)),
        filteredPosts: state.filteredPosts.filter(post => !state.selectedPosts.includes(post.id)),
        selectedPosts: [],
      }
    case SELECT_POST:
      return {
        ...state,
        selectedPosts: state.selectedPosts.includes(payload)
          ? state.selectedPosts.filter(postId => payload !== postId)
          : [...state.selectedPosts, payload],
      }

    case SET_POSTS_ERROR:
      return { ...state, error: payload }
    case SET_POSTS_LOADING:
      return { ...state, isLoading: payload }
    case ORDER_BY:
      return {
        ...state,
        selectedPosts: [],
        orderBy: payload,
        filteredPosts: sortingAndFilteringPosts(state.filteredPosts, payload, ''),
      }
    case FILTER:
      return {
        ...state,
        filter: payload,
        filteredPosts: sortingAndFilteringPosts(state.posts, state.orderBy, payload),
      }
    default:
      return state
  }
}
export default blog
