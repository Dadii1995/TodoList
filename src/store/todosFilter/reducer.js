import { VisibilityFilters } from './actions'
import { SET_VISIBILITY_FILTER, FILTER_TODOS } from './types'

export const initialState = { byStatus: VisibilityFilters.SHOW_ALL, byName: '' }

const todosFilter = (state = initialState, action) => {
  switch (action.type) {
    case SET_VISIBILITY_FILTER: {
      return { ...state, byStatus: action.payload }
    }
    case FILTER_TODOS: {
      return { ...state, byName: action.payload }
    }
    default:
      return state
  }
}

export default todosFilter
