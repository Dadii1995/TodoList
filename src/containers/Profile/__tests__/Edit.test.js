import React from 'react'
import renderer, { act } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

import Edit from '../Edit'
import { initialState } from '../../../store/profile/reducer'

const getWrapper = initialStore => {
  const mockStore = configureStore()
  const store = mockStore(initialStore)
  const wrapper = renderer.create(
    <Provider store={store}>
      <Edit />
    </Provider>,
  )
  const root = wrapper.root

  return { root, wrapper, store }
}
describe('<Edit />', () => {
  test('renders editForm', () => {
    const { wrapper } = getWrapper({ profile: initialState })
    expect(wrapper).toMatchSnapshot()
  })

})
