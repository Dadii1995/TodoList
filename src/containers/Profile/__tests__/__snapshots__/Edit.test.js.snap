// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edit /> renders editForm 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      First name
    </label>
    <input
      aria-label="First name"
      autoComplete="off"
      className="form-control"
      data-cy="user-name-input"
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="NewUser"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Photo url
    </label>
    <input
      aria-label="Photo url"
      autoComplete="off"
      className="form-control"
      data-cy="user-photo-input"
      name="url"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="http://examplewebsite.com/img/1.png"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Birthday
    </label>
    <input
      aria-label="Birthday"
      autoComplete="off"
      className="form-control"
      data-cy="user-birthday-input"
      name="birthday"
      onBlur={[Function]}
      onChange={[Function]}
      type="date"
      value="1995-06-23"
    />
  </div>
  <button
    aria-label={null}
    className="btn btn-success btn-block"
    data-cy="submit-button"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Save
  </button>
</form>
`;
